src/rotation.jl

```diff
@@ -0,0 +1,290 @@
+    return yrot, xrot, yrot_f, xrot_f, yrot_int, xrot_int
+end
+
+
+"""
+   _bilinear_helper(yrot, xrot, yrot_f, xrot_f, yrot_int, xrot_int) 
+
+Some helper variables
+"""
+@inline function _bilinear_helper(yrot, xrot, yrot_f, xrot_f)
+    xdiff = (xrot - xrot_f)
+    xdiff_1minus = 1 - xdiff
+    ydiff = (yrot - yrot_f)
+    ydiff_1minus = 1 - ydiff
+    
+    return ydiff, ydiff_1minus, xdiff, xdiff_1minus
+end
+
+
+"""
+    _prepare_imrotate(arr, θ, rotation_center)
+
+Prepate `sin` and `cos`, creates the output array and converts type
+of `rotation_center` if required.
+"""
+function _prepare_imrotate(arr::AbstractArray{T}, θ, rotation_center) where T
+    # needed for rotation matrix
+    θ = mod(real(T)(θ), real(T)(2π))
+    rotation_center = real(T).(rotation_center)
+    sinθ, cosθ = sincos(real(T)(θ)) 
+    out = similar(arr)
+    fill!(out, 0)
+    return sinθ, cosθ, rotation_center, out
+end
+
+
+"""
+    _check_trivial_rotations!(out, arr, θ, rotation_center) 
+
+When `θ = 0 || π /2 || π || 3/2 || π` and if `rotation_center` 
+is in the middle of the array.
+For an even array of size 4, the rotation_center would need to be 2.5.
+For an odd array of size 5, the rotation_center would need to be 3.
+
+In those cases, rotations are trivial just by reversing or swapping some axes.
+"""
+function _check_trivial_rotations!(out, arr, θ, rotation_center; adjoint=false)
+    if iszero(θ)
+        out .= arr
+        return true 
+    end
+    # check for special cases where rotations are trivial
+    if (iseven(size(arr, 1)) && iseven(size(arr, 2)) && 
+        rotation_center[1] ≈ size(arr, 1) ÷ 2 + 0.5 && rotation_center[2] ≈ size(arr, 2) ÷ 2 + 0.5) ||
+        (isodd(size(arr, 1)) && isodd(size(arr, 2)) && 
+        (rotation_center[1] == size(arr, 1) ÷ 2 + 1 && rotation_center[1] == size(arr, 2) ÷ 2 + 1))
+        if θ ≈ π / 2 
+            if adjoint == false
+                out .= reverse(PermutedDimsArray(arr, (2, 1, 3, 4)), dims=(2,))
+            else
+                out .= reverse(PermutedDimsArray(arr, (2, 1, 3, 4)), dims=(1,))
+            end
+            return true
+        elseif θ ≈ π
+            out .= reverse(arr, dims=(1,2))
+            return true
+        elseif θ ≈ 3 / 2 * π
+            if adjoint == false
+                out .= reverse(PermutedDimsArray(arr, (2, 1, 3, 4)), dims=(1,))
+            else
+                out .= reverse(PermutedDimsArray(arr, (2, 1, 3, 4)), dims=(2,))
+            end
+            return true
+        end
+    end
+
+    return false
+end
+
+
+"""
+    imrotate(arr::AbstractArray{T, 4}, θ; method=:bilinear, rotation_center=size(arr) .÷ 2 .+ 1)
+
+Rotates an array in the first two dimensions around the center pixel `rotation_center`. 
+`rotation_center` is defined such that there is a real center pixel for even and odd values which is rotated around.
+For an even sized array of size `(4,4)` this would be `(3,3)`, for an odd array of size `(3,3)` this would be `(2,2)`
+
```

Comment:
I think the documentation could be a bit clearer regarding the rotation center. You say `rotation_center` is defined such that there is a real pixel. Does that mean that the user *has to* supply the rotation center in this way? Or is it possible to rotate around any float 2D coords?

Reply:
It rotates around any given position. But the default is a integer center pixel.
[1,2,3,4] would be around the 3. and [1,2,3] around the 2.

Reply:
Now it is clearer, thank you!